### PR TITLE
HPCC-24003 Clean up some coverity issues

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: dali/base/dasds.cpp [hunk 1]
```diff
@@ -4832,9 +4832,11 @@ IPropertyTree *loadStore(const char *storeFilename, IPTreeMaker *iMaker, unsigne
 // Not really coalescing, blocking transations and saving store (which will delete pending transactions).
 class CLightCoalesceThread : implements ICoalesce, public CInterface
 {
-    bool stopped, within24;
+    bool stopped = false;
+    bool within24 = false;
     Semaphore sem;
-    unsigned writeTransactionsNow, lastSaveWriteTransactions, lastWarning;
```
Comment (ghalliday) [inline dali/base/dasds.cpp:4837]: @jakesmith writeTransactionsNow not used - there is a variable with the same name inside a function

File: ecl/hqlcpp/hqlinline.cpp [hunk 2]
```diff
@@ -1820,9 +1820,6 @@ AliasKind ClassEvalContext::evaluateExpression(BuildCtx & ctx, IHqlExpression *
         }
     }
 
-    if (!evaluateLocally)
```
Comment (ghalliday) [inline ecl/hqlcpp/hqlinline.cpp:1823]: can never be reached - see code above

File: esp/platform/cumulativetimer.hpp [hunk 3]
```diff
@@ -41,11 +41,6 @@ class CumulativeTimer : public CInterface
     {
     }
 
-    CumulativeTimer(const CumulativeTimer& t)
```
Comment (ghalliday) [inline esp/platform/cumulativetimer.hpp:44]: @afishbeck this wasn't called, and is the same as the default copy constructor, except it left logLevel uninitialized.

File: esp/platform/espprotocol.hpp [hunk 4]
```diff
@@ -163,8 +163,7 @@ class esp_http_decl CEspProtocol : public CInterface,
     CApplicationPortMap m_portmap;
     bool m_viewConfig;
     int m_MaxRequestEntityLength;
-    IEspContainer *m_container;
-    unsigned m_nextSeq;
```
Comment (ghalliday) [inline esp/platform/espprotocol.hpp:167]: @afishbeck was unused.

File: esp/platform/espsession.ipp [hunk 5]
```diff
@@ -33,7 +33,7 @@ private:
     StringAttr m_commentURL;
     StringAttr m_domain;
     StringAttr m_ports;
-    int        m_maxage;
+    int        m_maxage = 0;
```
Comment (ghalliday) [inline esp/platform/espsession.ipp:36]: @afishbeck could this have caused any problems?

**Review by ghalliday — COMMENTED** [review]
Please review

File: esp/platform/cumulativetimer.hpp (hunk 3 above)
Comment (afishbeck) [reply esp/platform/cumulativetimer.hpp:44]: looks fine

File: esp/platform/espsession.ipp (hunk 5 above)
Comment (afishbeck) [reply esp/platform/espsession.ipp:36]: If the uninitialized value happened to be negative then the cookie would be kept until browser closed.  A positive value would be time to live for the cookie.  Zero doesn't get set.  Odds are unless the value was a fairly small positive value we wouldn't have noticed.

File: dali/base/dasds.cpp [hunk 6]
```diff
@@ -8681,6 +8684,7 @@ bool CCovenSDSManager::fireException(IException *e)
         }
     };
     unhandledThread.setown(new CHandleException(*this, e, restartOnError));
+#endif
     return true;
```
Comment (richardkchapman) [inline dali/base/dasds.cpp:8688]: Doesn't this return true need to be inside the #if (or the one above removed)?

File: dali/server/daldap.cpp [hunk 7]
```diff
@@ -148,8 +148,8 @@ class CDaliLdapConnection: implements IDaliLdapConnection, public CInterface
         Owned<ISecUser> user = ldapsecurity->createUser(username);
         user->setAuthenticateStatus(AS_AUTHENTICATED);
 
-        bool filescope = stricmp(key,"Scope")==0;
-        bool wuscope = stricmp(key,"workunit")==0;
+        bool filescope = key && stricmp(key,"Scope")==0;
```
Comment (richardkchapman) [inline dali/server/daldap.cpp:151]: use strieq perhaps?

Comment (richardkchapman) [reply dali/server/daldap.cpp:151]: (perhaps strieq should be made safe!)

**Comment by richardkchapman** [comment]
@ghalliday One minor comment and one (probably incorrect) suggestion

File: dali/server/daldap.cpp (hunk 7 above)
Comment (ghalliday) [reply dali/server/daldap.cpp:151]: possibly, but I was aiming for minimal changes.  Not convinced about strieq.

File: dali/base/dasds.cpp (hunk 6 above)
Comment (ghalliday) [reply dali/base/dasds.cpp:8687]: moved inside.

**Comment by ghalliday** [comment]
@richardkchapman moved the return, but left the rest as it was

**Pull request merged**